apps/prairielearn/src/middlewares/selectAndAuthzAssessment.html.ts

```diff
@@ -0,0 +1,40 @@
+import { html } from '@prairielearn/html';
+
+import { HeadContents } from '../components/HeadContents.html.js';
+import { Navbar } from '../components/Navbar.html.js';
+import { config } from '../lib/config.js';
+
+export function AccessDenied({ resLocals }: { resLocals: Record<string, any> }) {
+  return html`
+    <!doctype html>
+    <html lang="en">
+      <head>
+        ${HeadContents({ resLocals })}
+      </head>
+      <body>
+        ${Navbar({ resLocals, navPage: 'assessment_instance' })}
+        <main id="content" class="container">
+          <div class="card mb-4">
+            <div class="card-header bg-primary text-white">Assessment unavailable</div>
+            <div class="card-body">
+              <p>This assessment's configuration does not allow you to access it right now.</p>
+              <p>This is the intended behavior based on your current time or other factors.</p>
+              <p>
+                <a
+                  href="/pl/course_instance/${resLocals.course_instance.id}/assessments"
```

Comment:
Nit: instructors shouldn't typically see this page after following a link, but they can if they end up at a bad URL, e.g. `http://localhost:3000/pl/course_instance/1/instructor/assessment/111111111111111111/questions`. If they then click this button, they'll be dropped into student mode, which could be unexpected.

It's unlikely that instructors will actually hit this case, so I'm fine if we don't do anything about it.